execution/map.go

```diff
@@ -64,21 +64,39 @@ func (stream *MappedStream) Next(ctx context.Context) (*Record, error) {
 		return nil, errors.Wrap(err, "couldn't get source record")
 	}
 
-	variables, err := stream.variables.MergeWith(srcRecord.AsVariables())
+	recordVariables := srcRecord.AsVariables()
+
+	variables, err := stream.variables.MergeWith(recordVariables)
 	if err != nil {
 		return nil, errors.Wrap(err, "couldn't merge given variables with record variables")
 	}
 
 	fieldNames := make([]octosql.VariableName, 0)
 	outValues := make(map[octosql.VariableName]octosql.Value)
 	for _, expr := range stream.expressions {
-		fieldNames = append(fieldNames, expr.Name())
+		switch expr := expr.(type) {
+		case *StarExpression:
+			// A star expression is based only on the incoming record
+			expressionName := expr.Fields(recordVariables)
+
+			for _, name := range expressionName {
+				value, _ := variables.Get(name) //TODO: err is always nil, handle it anyways?
+
+				outValues[name] = value
+			}
 
-		value, err := expr.ExpressionValue(ctx, variables)
-		if err != nil {
-			return nil, errors.Wrapf(err, "couldn't get expression %v", expr.Name())
+			fieldNames = append(fieldNames, expressionName...)
```

Comment:
In `default` case you first append `fieldNames` and then set `outValues` so I think you could move this line above for loop